### PR TITLE
docs(store): improve info about using lambdas in static methods

### PR DESCRIPTION
**Pull request by markwhitfeld**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #762 
Other PR: #766 

## What is the new behavior?

Improved grammar issues, moved to new section at the bottom because it pertains to all selectors that contain a lambda in the `return` statement, not just dynamic selectors and corrected alternative fix after some investigation.

### CODE REVIEW

File: docs/concepts/select.md [hunk 1]
```diff
@@ -331,3 +294,85 @@ export class CityService {
 ```
 
 now we can use this `zooThemeParks` selector anywhere in our application.
+
+## Special Considerations
+
+### Angular Libraries: Use of lambdas in static functions
+
+_If you are building an Angular lib directly so that it can be deployed to npm. `strictMetadataEmit` will most likely be set to true in the Angular compiler options and, as a result, Angular's `MetadataCollector` from the `@angular/compiler-cli` package will report the following issue with using lambdas in static methods:_
```
Comment (eranshmil) [inline docs/concepts/select.md]: What about linking to https://angular.io/guide/aot-compiler#strictmetadataemit?

File: docs/concepts/select.md [hunk 2]
```diff
@@ -331,3 +294,85 @@ export class CityService {
...
+
+>  Metadata collected contains an error that will be reported at runtime: Lambda not supported.`
+
+This error would be reported for each of the selectors defined below but, as demonstrated in the sample, you can prevent this by including the `// @dynamic` comment before the class expression and decorators:
```
Comment (eranshmil) [inline docs/concepts/select.md]: Btw, I think this comment also supported when placed above the selector itself.

Comment (markwhitfeld) [reply docs/concepts/select.md]: Didn't seem to work in my tests. I read that it needs to be above an exported symbol. ie. above the declaration of something that has an associated `export`.

File: docs/concepts/select.md (hunk 1 above)
Comment (markwhitfeld) [reply docs/concepts/select.md]: Good point. Done 👍

**Comment by markwhitfeld** [comment]
cc @splincode and @arturovt 
Note the corrected details on this issue

**Pull request merged**